sim/global-types.ts

```diff
@@ -24,8 +24,14 @@ type PokemonSlot = '' | string & {__isSlot: true};
 interface AnyObject {[k: string]: any}
 
 type GenderName = 'M' | 'F' | 'N' | '';
-type StatIDExceptHP = 'atk' | 'def' | 'spa' | 'spd' | 'spe';
-type StatID = 'hp' | StatIDExceptHP;
+const _statIDsExceptHP = ['atk', 'def', 'spa', 'spd', 'spe'] as const;
+const _statIDs = [..._statIDsExceptHP, 'hp'] as const;
+namespace Stats {
+	export const statIDsExceptHP = _statIDsExceptHP;
+	export const statIDs = _statIDs;
+}
+type StatIDExceptHP = typeof _statIDsExceptHP[number];
```

Comment:
Re-defining in terms of array so only need to change the stat list in one place. `as const` changes the array into a tuple, so that `StatIDExceptHP` is the type union of the elements of `_statIDsExceptHP`, i.e. `'atk' | 'def' | 'spa' | 'spd' | 'spe'`, just as before.

Reply:
Do these need to be defined as `const`s and then added to the namespace? Can't you just say 
```
namespace Stats {
	export const statIDsExceptHP = ['atk', 'def', 'spa', 'spd', 'spe'] as const;
...
```

Reply:
If you define them in the namespace, you can't use them outside of it for `StatIDExceptHP`, etc. I did it this way so the arrays in the namespace and the types outside it could be in sync. We _could_ move `StatID` into the namespace, but then we'd need to refactor a lot of imports, which I wanted to avoid.